server/channels/app/sso.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"crypto/sha256"
+	"crypto/subtle"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func (a *App) CreateCodeChallengeToken(codeChallenge string) (*model.Token, *model.AppError) {
+	extraProps, extraErr := json.Marshal(map[string]string{"code_challenge": codeChallenge})
+	if extraErr != nil {
+		return nil, model.NewAppError("App.CreateCodeChallengeToken", "api.oauth.store_code_challenge", nil, "", http.StatusInternalServerError).Wrap(extraErr)
+	}
+	token := model.NewToken("pkce", string(extraProps))
```

Comment:
3/5 I think we should get rid of the references to PKCE because this really has very little to do with PKCE even if it is PKCE-inspired.